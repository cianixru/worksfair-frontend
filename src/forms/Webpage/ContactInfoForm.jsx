/* eslint-disable react/jsx-no-bind */
/* eslint-disable jsx-a11y/accessible-emoji */
import React, { Component } from 'react';
import { Form, Field } from 'react-final-form';
import PropTypes from 'prop-types';

import WebpageText from '../../atoms/WebpageText';
import { states } from '../../utils/helpers';

class ContactInfoForm extends Component {
  state = {
    towns: null,
    currentState: '',
  }

  /**
   * @description this handles the selection of state, sets the towns
   * sets the state with the right values
   *
   * @param { object } event - DOM event
  */
  handleStateSelection = (event) => {
    event.preventDefault();
    const { value } = event.target;
    this.setState({
      towns: states[value],
      currentState: value,
    });
  }

  render() {
    const {
      onSubmit,
      validate,
      validationErrors,
      handleErrorReset,
    } = this.props;
    const { towns, currentState } = this.state;

    return (
      <Form
        onSubmit={onSubmit}
        validate={validate}
        // eslint-disable-next-line react/jsx-no-bind
        render={({
          handleSubmit, pristine, invalid, values
        }) => {
          return (
            <form onSubmit={handleSubmit}>
              <div className="field-body">
                <div className="field">
                  <h5>State</h5>
                  <div className="control">
                    <div className="select is-fullwidth">
                      <Field
                        name="state"
                      >
                        {({ input }) => {
                          values.state = currentState;
                          return (
                            <select
                              {...input}
                              value={currentState}
                              onChange={this.handleStateSelection}
                            >
                              <option>Select State</option>
                              {
                                Object.keys(states).map(elem => (
                                  <option
                                    value={elem}
                                    key={elem}>{elem}</option>
                                ))
                              }
                            </select>);
                        }}
                      </Field>
                    </div>
                  </div>
                </div>

                <div className="field">
                  <h5>Town</h5>
                  <div className="control">
                    <div className="select is-fullwidth">
                      <Field name="city" component="select">
                        <option>Select Town</option>
                        {
                          towns && towns.map(town => (
                            <option value={town} key={town}>{town}</option>
                          ))
                        }
                      </Field>
                    </div>
                  </div>
                </div>
              </div>

              <div className="field">
                <h5>Address</h5>
                <div className="control">
                  <WebpageText
                    className="input is-medium"
                    name="address"
                    validate={validate}
                    placeholder="Eg. No 12 ABC street, Zik Estate"
                    dataTestId="business-address"
                    required
                    validationErrors={validationErrors}
                    handleErrorReset={handleErrorReset}
                  />
                </div>
              </div>

              <div className="field-body">
                <div className="field">
                  <h5>Business Phone number</h5>
                  <div className="control">
                    <WebpageText
                      className="input is-medium"
                      name="phone"
                      validate={validate}
                      dataTestId="business-phone"
                      validationErrors={validationErrors}
                      handleErrorReset={handleErrorReset}
                      placeholder="08123456789"
                    />
                  </div>
                </div>

                <div className="field">
                  <h5>Business Email</h5>
                  <div className="control">
                    <WebpageText
                      className="input is-medium"
                      name="email"
                      validate={validate}
                      dataTestId="business-email"
                      validationErrors={validationErrors}
                      handleErrorReset={handleErrorReset}
                      placeholder="abc@anywebsite.com"
                    />
                  </div>
                </div>

                <div className="field">
                  <h5>Business Website
                    <span className="has-text-grey-light">
                    (if available)
                    </span>
                  </h5>
                  <div className="control">
                    <WebpageText
                      className="input is-medium"
                      name="website"
                      validate={validate}
                      dataTestId="business-website"
                      validationErrors={validationErrors}
                      handleErrorReset={handleErrorReset}
                      placeholder="https://www.yourwebsite.com"
                    />
                  </div>
                </div>
              </div>

              <div className="control">
                <button
                  type="submit"
                  disabled={pristine || invalid}
                  className="button is-danger is-medium is-rounded"
                  data-testid="update-contact"
                >
                  Save & Continue
                </button>
              </div>
            </form>
          );
        }}
      />
    );
  }
}

ContactInfoForm.propTypes = {
  onSubmit: PropTypes.func,
  validate: PropTypes.func,
  validationErrors: PropTypes.object,
  handleErrorReset: PropTypes.func,
};

export default ContactInfoForm;
