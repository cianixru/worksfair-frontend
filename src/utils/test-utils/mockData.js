// eslint-disable-next-line import/prefer-default-export
export const user = {
  username: 'Theookafor2',
  first_name: 'eod',
  last_name: 'okafor',
  email: 'theodorehope@andela.com',
  facebook: null,
  instagram: null,
  twitter: null,
  headline: null,
  current_location: null,
  role: 'admin',
  confirmed_account: false,
  verified_account: false,
  reset_token: null,
  image_url: null,
  created_at: '2019-03-10T08:48:42.942532Z',
  modified_at: '2019-03-10T08:48:42.942544Z',
  token: 'eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9',
};

export const webpage = {
  owner: {
    id: 1,
    username: 'theowa',
    first_name: 'theo',
    last_name: 'IO',
    email: 'theo.iowa@yahoo.com',
    facebook: null,
    instagram: null,
    twitter: null,
    headline: null,
    current_location: null,
    role: 'admin',
    confirmed_account: false,
    verified_account: false,
    reset_token: null,
    image_url: 'https://res.cloudinary.com/lbgcghcovyvtdbpih8ba.png',
    created_at: '2019-04-25T09:07:08.209916Z',
    modified_at: '2019-04-25T09:07:08.209930Z'
  },
  title: 'Happiness',
  sub_domain_name: 'happiness',
  description: 'Happiness truly lives here. We just happy',
  keywords: 'passion, software, engineering, technology, worksfair',
  country: 'Nigeria',
  state: 'Lagos',
  city: 'Okota',
  area: 'Omagba Phase 1, Phase 2, Abatete',
  address: 'No 7 Obinna Uzor Street, Canal Estate',
  phone: null,
  email: null,
  website: null,
  featured_images: [
    'https://res.cloudinary.com/lbgcghcovyvtdbpih8ba.png',
    'https://res.cloudinary.com/lbgcghcovyvtdbpih8ba.png'
  ],
  is_active: true,
  logo: null,
  colour: 'is-dark',
  created_at: '2019-04-28T18:39:25.174272Z',
  updated_at: '2019-05-01T12:18:49.669633Z'
};
