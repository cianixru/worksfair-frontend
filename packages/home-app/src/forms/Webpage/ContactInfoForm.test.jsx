import React from 'react';
import { render, cleanup, fireEvent } from 'react-testing-library';
import { BrowserRouter as Router } from 'react-router-dom';

import ContactInfoForm from './ContactInfoForm';
import { webpage } from '../../utils/test-utils/mockData';

afterEach(cleanup);

const validationErrors = {
  state: ['This field is required'],
  city: ['This field is required'],
  address: ['This field is required'],
  phone: ['This field is required'],
  email: ['This field is required'],
  website: ['This field is required'],
};

describe('ContactInfoForm.jsx', () => {
  const onSubmit = jest.fn();
  test('Test that the ContactInfo form displays', () => {
    const component = render(
      <Router>
        <ContactInfoForm
          onSubmit={onSubmit}
          validationErrors={validationErrors}
          webpage={webpage} />
      </Router>
    );
    expect(component).toMatchSnapshot();
  });

  test('should handle click for submission', () => {
    const handleClick = jest.fn();
    const { getByTestId } = render(
      <Router>
        <ContactInfoForm
          onSubmit={handleClick}
          validationErrors={validationErrors}
          webpage={webpage} />
      </Router>
    );
    const submitBtn = getByTestId('update-contact');
    submitBtn.onclick = handleClick;
    fireEvent.click(submitBtn);

    expect(handleClick).toHaveBeenCalledTimes(1);
  });
  test('should handle selection', () => {
    const handleClick = jest.fn();
    const { getByTestId } = render(
      <Router>
        <ContactInfoForm
          onSubmit={handleClick}
          validationErrors={validationErrors}
          webpage={webpage} />
      </Router>
    );
    const stateSelector = getByTestId('business-state');
    const townSelector = getByTestId('business-town');
    // stateSelector.onclick = handleClick;
    // townSelector.onclick = handleClick;
    fireEvent.select(stateSelector, { target: { value: 'Borno' }, });
    fireEvent.select(townSelector, { target: { value: 'Bama' }, });

    expect(stateSelector).toBeDefined();
    expect(townSelector).toBeDefined();
  });
});
