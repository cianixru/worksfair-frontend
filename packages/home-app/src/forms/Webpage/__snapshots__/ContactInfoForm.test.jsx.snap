// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactInfoForm.jsx Test that the ContactInfo form displays 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form>
        <div
          class="field-body"
        >
          <div
            class="field"
          >
            <h5>
              State
            </h5>
            <div
              class="control"
            >
              <div
                class="select is-fullwidth"
              >
                <select
                  data-testid="business-state"
                  name="state"
                >
                  <option>
                    Select State
                  </option>
                  <option
                    value="Abia"
                  >
                    Abia
                  </option>
                  <option
                    value="Adamawa"
                  >
                    Adamawa
                  </option>
                  <option
                    value="AkwaIbom"
                  >
                    AkwaIbom
                  </option>
                  <option
                    value="Anambra"
                  >
                    Anambra
                  </option>
                  <option
                    value="Bauchi"
                  >
                    Bauchi
                  </option>
                  <option
                    value="Bayelsa"
                  >
                    Bayelsa
                  </option>
                  <option
                    value="Benue"
                  >
                    Benue
                  </option>
                  <option
                    value="Borno"
                  >
                    Borno
                  </option>
                  <option
                    value="CrossRiver"
                  >
                    CrossRiver
                  </option>
                  <option
                    value="Delta"
                  >
                    Delta
                  </option>
                  <option
                    value="Ebonyi"
                  >
                    Ebonyi
                  </option>
                  <option
                    value="Edo"
                  >
                    Edo
                  </option>
                  <option
                    value="Ekiti"
                  >
                    Ekiti
                  </option>
                  <option
                    value="Enugu"
                  >
                    Enugu
                  </option>
                  <option
                    value="Gombe"
                  >
                    Gombe
                  </option>
                  <option
                    value="Imo"
                  >
                    Imo
                  </option>
                  <option
                    value="Jigawa"
                  >
                    Jigawa
                  </option>
                  <option
                    value="Kaduna"
                  >
                    Kaduna
                  </option>
                  <option
                    value="Kano"
                  >
                    Kano
                  </option>
                  <option
                    value="Katsina"
                  >
                    Katsina
                  </option>
                  <option
                    value="Kebbi"
                  >
                    Kebbi
                  </option>
                  <option
                    value="Kogi"
                  >
                    Kogi
                  </option>
                  <option
                    value="Kwara"
                  >
                    Kwara
                  </option>
                  <option
                    value="Lagos"
                  >
                    Lagos
                  </option>
                  <option
                    value="Nasarawa"
                  >
                    Nasarawa
                  </option>
                  <option
                    value="Niger"
                  >
                    Niger
                  </option>
                  <option
                    value="Ogun"
                  >
                    Ogun
                  </option>
                  <option
                    value="Ondo"
                  >
                    Ondo
                  </option>
                  <option
                    value="Osun"
                  >
                    Osun
                  </option>
                  <option
                    value="Oyo"
                  >
                    Oyo
                  </option>
                  <option
                    value="Plateau"
                  >
                    Plateau
                  </option>
                  <option
                    value="Rivers"
                  >
                    Rivers
                  </option>
                  <option
                    value="Sokoto"
                  >
                    Sokoto
                  </option>
                  <option
                    value="Taraba"
                  >
                    Taraba
                  </option>
                  <option
                    value="Yobe"
                  >
                    Yobe
                  </option>
                  <option
                    value="Zamfara"
                  >
                    Zamfara
                  </option>
                  <option
                    value="FCT"
                  >
                    FCT
                  </option>
                </select>
              </div>
            </div>
          </div>
          <div
            class="field"
          >
            <h5>
              Town
            </h5>
            <div
              class="control"
            >
              <div
                class="select is-fullwidth"
              >
                <select
                  data-testid="business-town"
                  name="city"
                >
                  <option>
                    Select Town
                  </option>
                  <option
                    value="Ikeja"
                  >
                    Ikeja
                  </option>
                  <option
                    value="Lekki"
                  >
                    Lekki
                  </option>
                  <option
                    value="Oshodi"
                  >
                    Oshodi
                  </option>
                  <option
                    value="Ikoyi"
                  >
                    Ikoyi
                  </option>
                  <option
                    value="Lagos-Island"
                  >
                    Lagos-Island
                  </option>
                  <option
                    value="Badagry"
                  >
                    Badagry
                  </option>
                  <option
                    value="Isolo"
                  >
                    Isolo
                  </option>
                  <option
                    value="Mushin"
                  >
                    Mushin
                  </option>
                  <option
                    value="Surulere"
                  >
                    Surulere
                  </option>
                  <option
                    value="Yaba"
                  >
                    Yaba
                  </option>
                </select>
              </div>
            </div>
          </div>
        </div>
        <div
          class="field"
        >
          <h5>
            Address
          </h5>
          <div
            class="control"
          >
            <input
              class="input is-medium"
              data-testid="business-address"
              name="address"
              placeholder="Eg. No 12 ABC street, Zik Estate"
              type="text"
              value="No 7 Obinna Uzor Street, Canal Estate"
            />
            <span
              class="input-error"
            >
              This field is required
            </span>
          </div>
        </div>
        <div
          class="field-body"
        >
          <div
            class="field"
          >
            <h5>
              Business Phone number
            </h5>
            <div
              class="control"
            >
              <input
                class="input is-medium"
                data-testid="business-phone"
                name="phone"
                placeholder="08123456789"
                type="text"
                value=""
              />
              <span
                class="input-error"
              >
                This field is required
              </span>
            </div>
          </div>
          <div
            class="field"
          >
            <h5>
              Business Email
            </h5>
            <div
              class="control"
            >
              <input
                class="input is-medium"
                data-testid="business-email"
                name="email"
                placeholder="abc@anywebsite.com"
                type="text"
                value=""
              />
              <span
                class="input-error"
              >
                This field is required
              </span>
            </div>
          </div>
          <div
            class="field"
          >
            <h5>
              Business Website
              <span
                class="has-text-grey-light"
              >
                (if available)
              </span>
            </h5>
            <div
              class="control"
            >
              <input
                class="input is-medium"
                data-testid="business-website"
                name="website"
                placeholder="https://www.yourwebsite.com"
                type="text"
                value=""
              />
              <span
                class="input-error"
              >
                This field is required
              </span>
            </div>
          </div>
        </div>
        <div
          class="control"
        >
          <button
            class="button is-link is-medium is-rounded"
            data-testid="update-contact"
            disabled=""
            type="submit"
          >
            Save & Continue
          </button>
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <form>
      <div
        class="field-body"
      >
        <div
          class="field"
        >
          <h5>
            State
          </h5>
          <div
            class="control"
          >
            <div
              class="select is-fullwidth"
            >
              <select
                data-testid="business-state"
                name="state"
              >
                <option>
                  Select State
                </option>
                <option
                  value="Abia"
                >
                  Abia
                </option>
                <option
                  value="Adamawa"
                >
                  Adamawa
                </option>
                <option
                  value="AkwaIbom"
                >
                  AkwaIbom
                </option>
                <option
                  value="Anambra"
                >
                  Anambra
                </option>
                <option
                  value="Bauchi"
                >
                  Bauchi
                </option>
                <option
                  value="Bayelsa"
                >
                  Bayelsa
                </option>
                <option
                  value="Benue"
                >
                  Benue
                </option>
                <option
                  value="Borno"
                >
                  Borno
                </option>
                <option
                  value="CrossRiver"
                >
                  CrossRiver
                </option>
                <option
                  value="Delta"
                >
                  Delta
                </option>
                <option
                  value="Ebonyi"
                >
                  Ebonyi
                </option>
                <option
                  value="Edo"
                >
                  Edo
                </option>
                <option
                  value="Ekiti"
                >
                  Ekiti
                </option>
                <option
                  value="Enugu"
                >
                  Enugu
                </option>
                <option
                  value="Gombe"
                >
                  Gombe
                </option>
                <option
                  value="Imo"
                >
                  Imo
                </option>
                <option
                  value="Jigawa"
                >
                  Jigawa
                </option>
                <option
                  value="Kaduna"
                >
                  Kaduna
                </option>
                <option
                  value="Kano"
                >
                  Kano
                </option>
                <option
                  value="Katsina"
                >
                  Katsina
                </option>
                <option
                  value="Kebbi"
                >
                  Kebbi
                </option>
                <option
                  value="Kogi"
                >
                  Kogi
                </option>
                <option
                  value="Kwara"
                >
                  Kwara
                </option>
                <option
                  value="Lagos"
                >
                  Lagos
                </option>
                <option
                  value="Nasarawa"
                >
                  Nasarawa
                </option>
                <option
                  value="Niger"
                >
                  Niger
                </option>
                <option
                  value="Ogun"
                >
                  Ogun
                </option>
                <option
                  value="Ondo"
                >
                  Ondo
                </option>
                <option
                  value="Osun"
                >
                  Osun
                </option>
                <option
                  value="Oyo"
                >
                  Oyo
                </option>
                <option
                  value="Plateau"
                >
                  Plateau
                </option>
                <option
                  value="Rivers"
                >
                  Rivers
                </option>
                <option
                  value="Sokoto"
                >
                  Sokoto
                </option>
                <option
                  value="Taraba"
                >
                  Taraba
                </option>
                <option
                  value="Yobe"
                >
                  Yobe
                </option>
                <option
                  value="Zamfara"
                >
                  Zamfara
                </option>
                <option
                  value="FCT"
                >
                  FCT
                </option>
              </select>
            </div>
          </div>
        </div>
        <div
          class="field"
        >
          <h5>
            Town
          </h5>
          <div
            class="control"
          >
            <div
              class="select is-fullwidth"
            >
              <select
                data-testid="business-town"
                name="city"
              >
                <option>
                  Select Town
                </option>
                <option
                  value="Ikeja"
                >
                  Ikeja
                </option>
                <option
                  value="Lekki"
                >
                  Lekki
                </option>
                <option
                  value="Oshodi"
                >
                  Oshodi
                </option>
                <option
                  value="Ikoyi"
                >
                  Ikoyi
                </option>
                <option
                  value="Lagos-Island"
                >
                  Lagos-Island
                </option>
                <option
                  value="Badagry"
                >
                  Badagry
                </option>
                <option
                  value="Isolo"
                >
                  Isolo
                </option>
                <option
                  value="Mushin"
                >
                  Mushin
                </option>
                <option
                  value="Surulere"
                >
                  Surulere
                </option>
                <option
                  value="Yaba"
                >
                  Yaba
                </option>
              </select>
            </div>
          </div>
        </div>
      </div>
      <div
        class="field"
      >
        <h5>
          Address
        </h5>
        <div
          class="control"
        >
          <input
            class="input is-medium"
            data-testid="business-address"
            name="address"
            placeholder="Eg. No 12 ABC street, Zik Estate"
            type="text"
            value="No 7 Obinna Uzor Street, Canal Estate"
          />
          <span
            class="input-error"
          >
            This field is required
          </span>
        </div>
      </div>
      <div
        class="field-body"
      >
        <div
          class="field"
        >
          <h5>
            Business Phone number
          </h5>
          <div
            class="control"
          >
            <input
              class="input is-medium"
              data-testid="business-phone"
              name="phone"
              placeholder="08123456789"
              type="text"
              value=""
            />
            <span
              class="input-error"
            >
              This field is required
            </span>
          </div>
        </div>
        <div
          class="field"
        >
          <h5>
            Business Email
          </h5>
          <div
            class="control"
          >
            <input
              class="input is-medium"
              data-testid="business-email"
              name="email"
              placeholder="abc@anywebsite.com"
              type="text"
              value=""
            />
            <span
              class="input-error"
            >
              This field is required
            </span>
          </div>
        </div>
        <div
          class="field"
        >
          <h5>
            Business Website
            <span
              class="has-text-grey-light"
            >
              (if available)
            </span>
          </h5>
          <div
            class="control"
          >
            <input
              class="input is-medium"
              data-testid="business-website"
              name="website"
              placeholder="https://www.yourwebsite.com"
              type="text"
              value=""
            />
            <span
              class="input-error"
            >
              This field is required
            </span>
          </div>
        </div>
      </div>
      <div
        class="control"
      >
        <button
          class="button is-link is-medium is-rounded"
          data-testid="update-contact"
          disabled=""
          type="submit"
        >
          Save & Continue
        </button>
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
